### DISCUSSION TITLE
Switch between "Pluto" reactive mode and "Jupyter" mode

### DISCUSSION THREAD

**Discussion by gianmariomanca** [fonsp/Pluto.jl#1344] [Pluto development]
This is related to the temporary execution barrier feature already merged to master.
It looks like there is already all the needed technology to be able to switch back and forth between the default Pluto reactive modality and the Jupyter style (i.e. user triggering the execution of selected cells as needed/desired) that can be convenient sometimes.

Basically in "Jupyter mode" you could simply highlight which cells are outdated. The same way you disable all dependent cells when you disable a cell, you could declare as outdated all cells that depend on a cell that has been re-run by the user. 

You could declare as outdated but runnable the immediate dependencies of the recently modified cell, and outdated and frozen all secondary, tertiary, etc dependencies. Basically guiding the user to follow step by step the correct dependency tree.

This would basically make Jupyter obsolete (ok maybe I'm exaggerating, there might be other Jupyter feature that are useful).

**Comment by gianmariomanca** [comment]
Ping. Just wanted to attract attention to this proposal. I believe it would be great during heavy development of large simulation/modeling notebooks with a lot of expensive cells. At least I find myself continuously wishing for it.
I would love to be in "Jupyter" mode (as described above) while changing various pieces here and there without re-triggering evaluation of the entire notebook and having to manually set up 20 different execution barriers. And then switch back to reactive mode once thing stabilize and I'm just testing config changes etc.

**Reply by kbarros** [reply]
My comment previously said this:

> Another vote for this feature. For notebooks with very long-to-execute cells, there is demand for a Jupyter-like experience. The Neptune.jl fork seems to provide this, but has not been keeping up with the Pluto.jl main branch.

My feelings on this have changed. Pluto now includes some features to allow the user to selectively introduce a sprinkling of mutable state where needed. For example, the PlutoHooks package has a powerful `@use_ref` macro  (https://juliapluto.github.io/PlutoHooks.jl/src/notebook.html). One can use this to store the state of a long-running simulation.  Here is an example of how to use this to make interactive, long-running simulations: https://github.com/fonsp/Pluto.jl/issues/476#issuecomment-1187834947

Another (relatively) new feature is the ability to selectively "disable" cells, if you want to pause execution of those cells and their downstream dependencies. Access this option via the three dots `...` on the far right of a cell. Image attached.

<img width="268" alt="image" src="https://user-images.githubusercontent.com/290205/179570102-31885d46-1c80-49c5-b45c-7a5a0c53a8f0.png">

**However**, I do believe there is a significant _discoverability_ problem. These features are not prominently advertised.

**Comment by bryaan** [comment]
Can we revisit making "Jupyter Mode" a reality?  For my use cases reactivity is simply broken.  I've discovered instances where there's definitely bugs.  And it's model is more complicated than jupyter which isn't for everyone.  

How long would it take to make this a reality as it's one thing setting me back from using Pluto seriously?

**Reply by pankgeorg** [reply]
You can check out the Neptune.jl project.

**Reply by bryaan** [reply]
Neptune.jl is dead.  We need Pluto to have this functionality.

**Reply by pankgeorg** [reply]
I'm sorry to hear that. If anything, we sympathize with the fact that maintaining a large open source project is a hard job and it's easy to get left behind.

Since Neptune.jl forked, there have been many discussions around non-reactive versions of Pluto. Unfortunately it is just out of scope for us, so we will not work on it. I'm sorry that it's setting you back from using Pluto seriously, but there is nothing I can do about it.

You might want to review @kbarros answer above ([permalink](https://github.com/fonsp/Pluto.jl/discussions/1344#discussioncomment-3070526)) for solutions  to your problem that respect Pluto's design decisions and scope, in a way that may seem unintuitive at first, but is quite powerful when you get ahead of it. If you want help with that, we will be happy to assist, in a setting where the [community standards are being](https://julialang.org/community/standards/) upheld (specifically citing _"a welcoming, civil and constructive environment")_ and _"ideas and bug reports are ok, but demands are not"_).

If that doesn't help either, it's probably best to give up on Pluto and use Jupyter or some other tool as they seem to be better choices for your use-case.

Best,
Panagiotis
